### PR TITLE
autobuild spiffs image from files in local/fs folder

### PR DESCRIPTION
**Pull request by DesperateProgrammer**
As the last try to PR, but now clean into one commit. Thank you @TerryE for the pointers. I guess it will still need a lot of excercise until I get a smooth usage of git ;-)

The files in local/fs will be taken to build thespiffs image.
The spiffs image is put into the bin folder with the name matching the destination address within the flash given the 0x10000.bin image.

For building the spiffsimg tool gcc is used (the xtensa compiler is overridden). This should make this Makefile usable everywhere gcc is installed and within the path.

For spiffsimg there is a bug in VirtualBox that keeps crashing spiffsimg with a segmentation fault (mmap is not supported in MAP_SHARED mode). This needs to be fixed within the spiffsimg git and is not part of this PR.

Refer [https://github.com/nodemcu/nodemcu-firmware/issues/1031]

### CODE REVIEW

**Comment by DesperateProgrammer** [comment]
Thank you.
Fixed it together with a stray ;) from a copy&paste

File: tools/Makefile [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+.NOTPARALLEL:
+ifndef PDIR
+
+endif
+#############################################################
+# Options
+#
+
+FSSOURCE ?= ../local/fs/
+FSSIZE ?= 196608
```
Comment (pjsg) [inline tools/Makefile]: Where does this number come from?

**Comment by marcelstoer** [comment]
@DesperateProgrammer can you please finish up here (squash commits and answer Philip's question) so we can merge.

**Comment by pjsg** [comment]
I have taken this code and merged it into my branch referred to in #1164

**Comment by marcelstoer** [comment]
Philip, you can close this one then, right?

File: docs/en/flash.md [hunk 2]
```diff
@@ -43,6 +43,8 @@ Otherwise, if you built your own firmware from source code:
 
 Also, in some special circumstances, you may need to flash `blank.bin` or `esp_init_data_default.bin` to various addresses on the flash (depending on flash size and type), see [below](#upgrading-from-sdk-09x-firmware).
 
+If you have files added into the `local/fs/` folder you might want to flash the file named in `bin/spiffs.dat` to the address it's named about. This will copy the filesystem created as a copy of `local/fs/` to your nodeMCU and made available by the lua file module.
```
Comment (marcelstoer) [inline docs/en/flash.md]: It should be *N*odeMCU and *L*ua (w/o italics of course).

File: local/fs/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+*
+!.gitignore
```
Comment (marcelstoer) [inline local/fs/.gitignore]: Missing a newline here.

I suggest to add comments to make the intention very clear. How about:

```
# Ignore everything
*
# But not this file itself.
!.gitignore
```

**Comment by pjsg** [comment]
I'm closing this one as I have merged it into the same branch as the new spiffs version.

**Pull request closed without merging**